In [15]:
import pandas as pd
import numpy as np
from scipy import stats

In [3]:
df=pd.read_csv('c4_epa_air_quality.csv')

This is an air quality index dataset Source-Google Advanced Data Analytics Certification Modified Dataset

In [4]:
df.head()

Unnamed: 0  date_local    state_name   county_name      city_name  \
0           0  2018-01-01       Arizona      Maricopa        Buckeye   
1           1  2018-01-01          Ohio       Belmont      Shadyside   
2           2  2018-01-01       Wyoming         Teton  Not in a city   
3           3  2018-01-01  Pennsylvania  Philadelphia   Philadelphia   
4           4  2018-01-01          Iowa          Polk     Des Moines   

                                     local_site_name   parameter_name  \
0                                            BUCKEYE  Carbon monoxide   
1                                          Shadyside  Carbon monoxide   
2  Yellowstone National Park - Old Faithful Snow ...  Carbon monoxide   
3                             North East Waste (NEW)  Carbon monoxide   
4                                          CARPENTER  Carbon monoxide   

    units_of_measure  arithmetic_mean  aqi  
0  Parts per million         0.473684    7  
1  Parts per million         0.263158    5  
2  Parts per million         0.111111    2  
3  Parts per million         0.300000    3  
4  Parts per million         0.215789    3

In [5]:
sample_data=df.sample(n=50,replace=False)

In [7]:
sample_data.head()

Unnamed: 0  date_local  state_name county_name             city_name  \
258         258  2018-01-01    Nebraska     Douglas                 Omaha   
78           78  2018-01-01    Maryland     Garrett           Grantsville   
5             5  2018-01-01      Hawaii    Honolulu         Not in a city   
26           26  2018-01-01  California      Orange              La Habra   
215         215  2018-01-01        Ohio    Cuyahoga  Warrensville Heights   

         local_site_name   parameter_name   units_of_measure  arithmetic_mean  \
258                  NaN  Carbon monoxide  Parts per million         0.421053   
78             Piney Run  Carbon monoxide  Parts per million         0.200000   
5                Kapolei  Carbon monoxide  Parts per million         0.994737   
26              La Habra  Carbon monoxide  Parts per million         0.673684   
215  Cleveland Near Road  Carbon monoxide  Parts per million         0.321053   

     aqi  
258    9  
78     2  
5     14  
26    13  
215    5

In [13]:
mean=sample_data.aqi.mean()
mean

6.16

How succusful is the evaluation can be told to the stakeholder in form of confidence intervals which accounts for uncertanity due to margin of error among others

confidencr levels and their respective z scores used for calculation

  Confidence Level   Z score
  
   90                              1.64 
   
   95                              1.96
   
   99                             2.57

We have selected a confidence level of 95 for this analysis

In [10]:
standard_deviation=sample_data.aqi.std()

In [11]:
standard_error=standard_deviation/len(sample_data)

In [12]:
margin_of_error= 1.96*standard_error
#margin of error= Zscore*standard_error

In [16]:
Confidence_Interval=stats.norm.interval(alpha=0.95,loc=mean,scale=margin_of_error)
# alpha is leve of confidence
#loc is sample mean
#scale is margin of error
#Confidence Interval[mean-margin_of_error,mean+margin_of_error]

In [17]:
Confidence_Interval

(5.7328284723052185, 6.587171527694782)

The AQI across the 600+ districts can be [5.7328284723052185, 6.587171527694782] with a confidence level of 95 and the values of AQI below 100 is considered to be satisfactory so government can do futher invetidation of the AQI and how to keep is consistent or make it better

for smaller datasets withs ample size below 30 we use t distribution